samples/reactjs-todo/client/components/journey/form.js

```diff
@@ -172,7 +191,32 @@ export default function Form({ action, bottomMessage, followUp, topMessage }) {
         <WebAuthn step={renderStep} setSubmissionStep={setSubmissionStep} />
       </>
     );
+  } else if (
+    renderStep.type === 'Step' &&
+    renderStep?.getCallbacksOfType(CallbackType.RedirectCallback).length > 0
+  ) {
+    FRAuth.redirect(renderStep);
+    return <Loading message="Redirecting ..." />;
   } else if (renderStep.type === 'Step') {
+    /** *********************************************************************
+     * SDK INTEGRATION POINT
+     * Summary: Handle selection of IdpCallbacks
+     * ----------------------------------------------------------------------
+     * Details: If we do not initially select the IdpCallback, when pressing enter,
+     * local authentication will automatically be submitted.
+     * This allows us to submit the form with the proper selection of an
+     * identity provider rather than local authentication.
+     ********************************************************************* */
+    const selectIdPCallback = renderStep?.getCallbacksOfType(CallbackType.SelectIdPCallback);
+
+    if (selectIdPCallback?.length > 0) {
+      const nameCallBacksInputs = renderStep?.getCallbackOfType(CallbackType.NameCallback);
+      const idToken2Input = nameCallBacksInputs?.getInputValue('IDToken2');
+
+      if (isIdentityProviderLogin(selectIdPCallback[0]) && idToken2Input !== '') {
+        selectIdPCallback[0].setInputValue('localAuthentication', 0);
+      }
+    }
```

Comment:
I'm not sure I understand the intent here. Can you elaborate on what this is doing? 

Just speaking generically, there are a few things that give me hesitation here:

1. It seems like we have logic at the form level that should be at the callback level. We should be able to achieve a cleaner implementation, similar to what we have in the Login Framework. Please checkout the implementation in that project to get a sense of how we do it there.
2. We have logic that uses values that may be inconsistent between journey designs. The reliance of `IDToken2` will not always be reliable as it's an implementation detail of the journey itself, rather than a guaranteed interface.

Reply:
thanks for the comment, let's see if i can be clear enough. When logging locally via user/password, the step coming back from AM comes along with 3 types of CB (selectIDPcallback, nameCallback and passwordCallback). I'd say due to the implementation of the journey/AM, the selectIdPCallback .input[0] for the payload has the value of the first idp, in this case facebook, so it leads to a failure authentication as if it is local log-in that value "selectIdPCallback --> input[0]  -->**idToken1**" must be "localAuthentication". So having said this, i proceed to do this exactly:
1) i check the existence of a selectIdpCallback, since this "misbehavior" only occurs for the selectIdpCallback
2) if so, grab for the nameCallback to detect if it's local auth. If this value is empty(the user didn't enter a name for that input in the form), it is social-login log-in, don't do anything.
3) Check for the the value of its input, if it has a value we want it to be 'localAuthentication' as we're logging in bia user/password **through a selectIdpCallback** , hence set the value manually. Keep in mind that it could be a selectIdpCallback and still be a user/password log-in, which is why i only set the value manually if nameCallback has a value. I could've logged in via google and nameCallback would be emtpy, as the user didn't enter the username, in that case i wouldn't do anything
I attach an image where i'm logging in via username/password, the stepRender has a value of "facebook-web" for the selectIdPCallback .input[0], as mentioned before it is supposed to be 'localAuthentiction', then i proceed to set it.
<img width="1377" alt="image" src="https://github.com/ForgeRock/forgerock-javascript-sdk/assets/18680830/9de022d2-efd3-460d-91a0-f2edd8b40a1a">

Reply:
UPDATE: we could go for this workaround. I took parte of the logic from the login widget but it's still necessary to check for the NameCallBack input. In the image, the nameCallbackInput is empty, which means it is accesing via idp(google-web-juan) and we don't need to set the provider to localAuthentication, we leave it as it is
<img width="1382" alt="image" src="https://github.com/ForgeRock/forgerock-javascript-sdk/assets/18680830/8ce8d9fd-69b8-4236-9393-12dd57be739d">

**solution proposed:**
<img width="686" alt="image" src="https://github.com/ForgeRock/forgerock-javascript-sdk/assets/18680830/a11ec2e4-5ffb-43b3-b3c0-aea8aadfcca0">

Reply:
> When logging locally via user/password, the step coming back from AM comes along with 3 types of CB (selectIDPcallback, nameCallback and passwordCallback). I'd say due to the implementation of the journey/AM, the selectIdPCallback .input[0] for the payload has the value of the first idp, in this case facebook, so it leads to a failure authentication as if it is local log-in that value "selectIdPCallback --> input[0] -->idToken1" must be "localAuthentication".

Sure, I believe the most optimal way of handling this is what we do in the `select-idp.svelte` component here:https://github.com/ForgeRock/forgerock-web-login-framework/blob/beta/src/lib/journey/callbacks/select-idp/select-idp.svelte#L50. 

The logic is simply this:

1. If the `SelectIdPCallback` has `localAuthentication` option, immediately change the default value to local option
2. If the user fills out the username and password without interaction of the providers, then no need to change from local option
3. If the user wants to use Google or whatever, then they will click on the provider altering the default we set
4. If the user clicks on on a social provider, they can't "change their mind" as the redirect will take them away from the Widget, and they will have to return, resetting the default back to local

Does that make sense? This way, callbacks don't need to know about each other and the stage level doesn't have to have knowledge about specific callbacks.

Reply:
i took the logic from this very component and if i only used that "if" the social provider log-in breaks. In the example below i logged in via google. i do have localAuthentication (1), therefore it will enter into the "if", and the provider(in thise case google-web-juan) (2) will be overwritten to "localAuthentication" but we dont want that, we want it to remain as "google-web-juan" as we're accesing via google. (4)
<img width="1286" alt="image" src="https://github.com/ForgeRock/forgerock-javascript-sdk/assets/18680830/75af49f6-10b2-40c5-8a8a-7f149574133e">

<img width="1286" alt="image" src="https://github.com/ForgeRock/forgerock-javascript-sdk/assets/18680830/dff49e8a-8995-4e60-895f-324361635d81">

Reply:
This is likely because you have this logic inside the "bare" component, rather than within a `useEffect` with `[]` dependencies. For every change in state, the components will rerun themselves, and you'll keep setting the provider back to `localAuthentication` over and over. What you want is to just set that value at first render, but NOT every subsequent render, allowing the user to select and keep the desired option.

Can you move this logic for setting the default `localAuthentication` provider setting into the `identity-provider` component, and wrap it in a `useEffect`, so it only runs once on initial render? I have a feeling that will fix your issue.

@ryanbas21 Does my above logic make sense to you. If it does, can you help out here?